prince-of-versions/src/main/java/co/infinum/princeofversions/JsonConfigurationParser.java

```diff
@@ -105,41 +106,86 @@ public PrinceOfVersionsConfig parse(String content) throws Throwable {
-                            builder.withMandatoryVersion(min);
-                        }
-                        if (update.has(LATEST_VERSION)) {
-                            int latest = update.getInt(LATEST_VERSION);
-                            builder.withOptionalVersion(latest);
-                        }
-                        if (update.has(NOTIFICATION)) {
-                            String notification = update.getString(NOTIFICATION);
-                            builder.withOptionalNotificationType(
-                                notification != null && notification.equalsIgnoreCase(NOTIFICATION_ALWAYS) ? NotificationType.ALWAYS
-                                    : NotificationType.ONCE);
-                        }
-                        break;
+            Object json = new JSONTokener(data.get(ANDROID).toString()).nextValue();
+            if (json instanceof JSONArray) {
+                JSONArray android = data.getJSONArray(ANDROID);
+                for (int i = 0; i < android.length(); i++) {
+                    JSONObject update = android.getJSONObject(i);
+                    if (parseJsonUpdate(update, builder, meta)) {
+                        return; //return after finding the first feasible update
                     }
                 }
+            } else if (json instanceof JSONObject) {
+                parseJsonUpdate(data.getJSONObject(ANDROID), builder, meta);
             }
         } else {
             throw new IllegalStateException("Config resource does not contain android key");
         }
-        if (data.has(META)) {
-            Object meta = data.get(META);
-            if (meta instanceof JSONObject) {
-                builder.withMetadata(jsonObjectToMap((JSONObject) meta));
+    }
+
+    private void mergeUpdateMetaWithDefaultMeta(Object meta, JSONObject update, PrinceOfVersionsConfig.Builder builder) throws
+        JSONException {
+        Object updateMeta;
+        if (meta == null) {
+            if (update.has(META)) {
+                updateMeta = update.get(META);
+                if (updateMeta instanceof JSONObject) {
+                    builder.withMetadata(jsonObjectToMap((JSONObject) updateMeta));
+                }
+            }
+        } else {
+            if (update.has(META)) {
+                updateMeta = update.get(META);
+                if (updateMeta instanceof JSONObject) {
+                    Map<String, String> metaMap = jsonObjectToMap((JSONObject) meta);
+                    Map<String, String> updateMetaMap = jsonObjectToMap((JSONObject) updateMeta);
+                    metaMap.putAll(updateMetaMap);
+                    builder.withMetadata(metaMap);
+                }
+            }
+        }
+    }
+
+    private void saveFirstAcceptableUpdate(JSONObject update, PrinceOfVersionsConfig.Builder builder) throws JSONException {
+        if (update.has(MINIMUM_VERSION)) {
+            int min = update.getInt(MINIMUM_VERSION);
```

Comment:
It would be great if we could check if this is int and if it isn't to throw our own exception with meaningful message, for example: "In update configuration $update required_version should be int, but the actual value is $update.get(MINIMUM_VERSION)".
It would make sense to do that for each property that we required certain type. That way user would be able to get clear message what is wrong with his json and where is the error.